subsys/net/lib/socks/socks.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2019 Antmicro Ltd
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(net_socks, CONFIG_SOCKS_LOG_LEVEL);
+
+#include <zephyr.h>
+#include <errno.h>
+#include <net/socket.h>
+#include <net/socks.h>
+
+#include "socks_internal.h"
+
+/**@brief Connects to destination through a SOCKS5 proxy server.
```

Comment:
So, this docstring should go to header. My understanding that API docs are generated from headers, or at least that's where we have docstrings in the rest of Zephyr.